### PR TITLE
feat: Lesson12 CSVDataLoader complete

### PR DESCRIPTION
**Pull request by RichHawkins**

### CODE REVIEW

File: lesson_12/io/io_app/src/main/java/com/codedifferently/lesson12/factory/richhawkins/CsvDataLoader.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import org.springframework.stereotype.Service;
+
+@Service
+public class CsvDataLoader implements LibraryCsvDataLoader {
+  @Override
+  public LibraryDataModel loadData() {
+    var model = new LibraryDataModel();
+    model.mediaItems =
+        readMediaItems(
+            "/workspaces/code-differently-24-q1/lesson_12/io/io_app/src/main/resources/csv/media_items.csv");
```
Comment (anthonydmays) [inline lesson_12/io/io_app/src/main/java/com/codedifferently/lesson12/factory/richhawkins/CsvDataLoader.java:28]: You need to grab the files from an app relative path. Check out what I did in `LibraryJsonDataLoader` to solve this.

**Review by anthonydmays — CHANGES_REQUESTED** [review]

File: lesson_12/io/io_app/src/main/java/com/codedifferently/lesson12/factory/richhawkins/CsvDataLoader.java (hunk 1 above)
Comment (RichHawkins) [reply lesson_12/io/io_app/src/main/java/com/codedifferently/lesson12/factory/richhawkins/CsvDataLoader.java]: Thanks, fixed this and a couple other things, all tests now pass!

File: lesson_12/io/io_app/src/main/java/com/codedifferently/lesson12/factory/richhawkins/CsvDataLoader.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+import org.springframework.stereotype.Service;
+
+@Service
+public class CsvDataLoader implements LibraryCsvDataLoader {
+  @Override
+  public LibraryDataModel loadData() {
+    var model = new LibraryDataModel();
+    model.mediaItems =
+        readMediaItems(
+            this.getClass().getClassLoader().getResource("csv/media_items.csv").getFile());
```
Comment (anthonydmays) [inline lesson_12/io/io_app/src/main/java/com/codedifferently/lesson12/factory/richhawkins/CsvDataLoader.java:28]: Oh I know you can clean this up...

**Review by anthonydmays — DISMISSED** [review]

File: lesson_12/io/io_app/src/main/java/com/codedifferently/lesson12/factory/richhawkins/CsvDataLoader.java (hunk 2 above)
Comment (RichHawkins) [reply lesson_12/io/io_app/src/main/java/com/codedifferently/lesson12/factory/richhawkins/CsvDataLoader.java]: Thanks for the push, code is much cleaner now.

**Review by anthonydmays — APPROVED** [review]

**Pull request merged**